src/threadvars.h

```diff
@@ -135,6 +135,10 @@ typedef struct ThreadVars_ {
     struct FlowQueue_ *flow_queue;
     bool break_loop;
 
+#ifdef HAVE_NDPI
+    struct ndpi_detection_module_struct *ndpi_struct;
```

Comment:
Hmm... I do not know about this one... (how we could make it work as a plugin)

Reply:
could just use `thread_local`?

Reply:
Would we delay initialization until first use, which would be in the packet path then?

I don't think a callback on thread creation would work, as in the library work a thread may be provided by the user, in which case some sort of callback with some user storage in the ThreadVar might make more sense?

Reply:
I'm fine with user data on ThreadVars. When/how to construct/destruct needs some thought. Do plugins have "thread init" callbacks? Where it would be called once per thread at init?

Reply:
> I'm fine with user data on ThreadVars. When/how to construct/destruct needs some thought. Do plugins have "thread init" callbacks? Where it would be called once per thread at init?

No they do not, but this is what I was thinking we might need.